nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -4337,4 +4474,487 @@ pub(crate) mod test {
 
         logctx.cleanup_successful();
     }
+
+    #[test]
+    fn test_sort_zones_by_deps() {
+        static TEST_NAME: &str = "sort_zones_by_deps";
+        let logctx = test_setup_log(TEST_NAME);
+        let log = logctx.log.clone();
+
+        // Collect zone configs from our example system.
+        let (_collection, _input, blueprint) = example(&log, TEST_NAME);
+        let mut zones = blueprint
+            .all_omicron_zones(BlueprintZoneDisposition::any)
+            .map(|(sled_id, z)| (sled_id, z.clone()))
+            .collect::<Vec<(SledUuid, BlueprintZoneConfig)>>();
+
+        // Sort them and verify the ordering constraints.
+        sort_zones_by_deps(&mut zones);
+        let mut nexus = false;
+        for kind in zones.iter().map(|(_, z)| z.zone_type.kind()) {
+            match kind {
+                ZoneKind::Nexus => {
+                    nexus = true;
+                }
+                _ => {
+                    assert!(!nexus);
+                }
+            }
+        }
+        assert!(nexus);
+
+        logctx.cleanup_successful();
+    }
+
+    /// Manually update the example system's inventory collection's zones
+    /// from a blueprint.
+    fn update_collection_from_blueprint(
+        example: &mut ExampleSystem,
+        blueprint: &Blueprint,
+    ) {
+        for (&sled_id, config) in blueprint.sleds.iter() {
+            example
+                .system
+                .sled_set_omicron_zones(
+                    sled_id,
+                    config.clone().into_in_service_sled_config().zones_config,
+                )
+                .expect("can't set omicron zones for sled");
+        }
+        example.collection =
+            example.system.to_collection_builder().unwrap().build();
+    }
+
+    #[test]
+    fn test_update_crucible_pantry() {
+        static TEST_NAME: &str = "update_crucible_pantry";
+        let logctx = test_setup_log(TEST_NAME);
+        let log = logctx.log.clone();
+
+        // Use our example system.
+        let mut rng = SimRngState::from_seed(TEST_NAME);
+        let (mut example, blueprint1) = ExampleSystemBuilder::new_with_rng(
+            &logctx.log,
+            rng.next_system_rng(),
+        )
+        .build();
+        verify_blueprint(&blueprint1);
+
+        // We should start with no specified TUF repo and nothing to do.
+        assert!(example.input.tuf_repo().is_none());
+        // assert_planning_makes_no_changes(
```

Comment:
Nit - can we uncomment this check? (Seems like it should be valid if there's no TUF repo yet, right?)

Reply:
We can indeed, that was left over from a debugging session where I didn't want the unchanged diff output. Uncommented in 2daabf512.